src/snowcli/cli/nativeapp/version/version_processor.py

```diff
@@ -0,0 +1,311 @@
+from pathlib import Path
+from sys import stdin, stdout
+from textwrap import dedent
+from typing import Dict, List, Optional
+
+import typer
+from git import Repo
+from git.exc import InvalidGitRepositoryError
+from rich import print
+from snowcli.cli.nativeapp.artifacts import find_version_info_in_manifest_file
+from snowcli.cli.nativeapp.constants import VERSION_COL
+from snowcli.cli.nativeapp.exceptions import ApplicationPackageDoesNotExistError
+from snowcli.cli.nativeapp.manager import (
+    NativeAppCommandProcessor,
+    NativeAppManager,
+    ensure_correct_owner,
+)
+from snowcli.cli.nativeapp.run_processor import NativeAppRunProcessor
+from snowcli.cli.nativeapp.utils import (
+    Prompts,
+    UserConfirmationPolicy,
+    ask_user_confirmation,
+    find_all_rows,
+    find_first_row,
+)
+from snowcli.cli.project.util import unquote_identifier
+from snowcli.exception import SnowflakeSQLExecutionError
+from snowflake.connector import ProgrammingError
+from snowflake.connector.cursor import DictCursor
+
+# Custom lambdas to plug in to UserConfirmationPolicy
+is_force_lambda = lambda force: force
+is_interactive_mode_lambda = lambda: stdin.isatty() and stdout.isatty()
+confirm_with_user_lambda = lambda prompt: typer.confirm(prompt)
+
+
+def check_index_changes_in_git_repo(
+    project_root: Path,
+    version: str,
+    package_name: str,
+    user_confirmation_policy: UserConfirmationPolicy,
+) -> None:
+    """
+    Checks if the project root, i.e. the native apps project is a git repository. If it is a git repository,
+    it also checks if there any local changes to the directory that may not be on the app package stage.
+    """
+    try:
+        repo = Repo(project_root)
```

Comment:
Not sure why we fold git repos into this, can you explain? Having local files that have never been synced is one thing, but not having files checked in is another. What if the repo is actually in a parent directory?

Reply:
This is implemented from our functional specs. Let's have that discussion in our channel.